server/mongo/src/storage.ts

```diff
@@ -434,7 +450,17 @@ abstract class MongoAdapterBase implements DbAdapter {
       enableUtf8Validation: false
     })
     cursor.maxTimeMS(30000)
-    const res = (await cursor.toArray())[0]
+    let res: Document = []
+    try {
+      res = (await cursor.toArray())[0]
+    } catch (e) {
+      console.error(
+        `error during executing cursor in findWithPipeline ${clazz} ${JSON.stringify(
```

Comment:
Better to return as separate console.error values, do not use JSON directly, since in debug it will be collapsed, and not as text.